server/src/test/java/org/elasticsearch/indices/SystemIndexDescriptorTests.java

```diff
@@ -263,66 +260,6 @@ private static String getVersionedMappings(int version) {
         return Strings.format(MAPPINGS_FORMAT_STRING, SystemIndexDescriptor.VERSION_META_KEY, version);
     }
 
-    public void testGetDescriptorCompatibleWith() {
-        final SystemIndexDescriptor prior = SystemIndexDescriptor.builder()
-            .setIndexPattern(".system*")
-            .setDescription("system stuff")
-            .setPrimaryIndex(".system-1")
-            .setAliasName(".system")
-            .setType(Type.INTERNAL_MANAGED)
-            .setSettings(Settings.EMPTY)
-            .setMappings(getVersionedMappings(TEST_MAPPINGS_PRIOR_VERSION))
-            .setVersionMetaKey("version")
-            .setOrigin("system")
-            .setMinimumNodeVersion(Version.V_7_0_0)
-            .build();
-        final SystemIndexDescriptor descriptor = SystemIndexDescriptor.builder()
-            .setIndexPattern(".system*")
-            .setDescription("system stuff")
-            .setPrimaryIndex(".system-1")
-            .setAliasName(".system")
-            .setType(Type.INTERNAL_MANAGED)
-            .setSettings(Settings.EMPTY)
-            .setMappings(MAPPINGS)
-            .setVersionMetaKey("version")
-            .setOrigin("system")
-            .setPriorSystemIndexDescriptors(List.of(prior))
-            .build();
-
-        SystemIndexDescriptor compat = descriptor.getDescriptorCompatibleWith(Version.CURRENT);
```

Comment:
I did not remove the test but updated it to use mappingVersions -- but I'm really fine either way